File: libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/OutgoingPushMessage.java
/**
 * Copyright (C) 2014-2016 Open Whisper Systems
 *
 * Licensed according to the LICENSE file in this repository.
 */

package org.whispersystems.signalservice.internal.push;


import com.fasterxml.jackson.annotation.JsonProperty;

public class OutgoingPushMessage {

  @JsonProperty
  public int    type;
  @JsonProperty
  public int    destinationDeviceId;
  @JsonProperty
  public int    destinationRegistrationId;
  @JsonProperty
  public String content;

  public OutgoingPushMessage() {}

  public OutgoingPushMessage(int type,
                             int destinationDeviceId,
                             int destinationRegistrationId,
                             String content)
  {
    this.type                      = type;
    this.destinationDeviceId       = destinationDeviceId;
    this.destinationRegistrationId = destinationRegistrationId;
    this.content                   = content;
  }

  public int getDestinationDeviceId() {
    return destinationDeviceId;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/OutgoingPushMessage.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

data class OutgoingPushMessage(
  val type: Int,
  val destinationDeviceId: Int,
  val destinationRegistrationId: Int,
  val content: String
)
